pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -175,7 +175,8 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration, featur
 		allErrors = append(allErrors, fmt.Errorf("invalid configuration: topologyManagerScope (--topology-manager-scope) %q must be one of: %q, or %q", kc.TopologyManagerScope, kubeletconfig.ContainerTopologyManagerScope, kubeletconfig.PodTopologyManagerScope))
 	}
 
-	if localFeatureGate.Enabled(features.GracefulNodeShutdown) {
+	gracefulNodeShutdownEnabled := localFeatureGate.Enabled(features.GracefulNodeShutdown) || localFeatureGate.Enabled(features.WindowsGracefulNodeShutdown)
```

Comment:
Does it mean that Graceful Node Shutdown will be activated on Linux even when `GracefulNodeShutdown` feature 
 is not enabled, but `WindowsGracefulNodeShutdown` is enabled?

Reply:
@bart0sh no, depends on which platform are you on. Once any one of above feature-gates is enabled, the validation will be executed and metrics will be collected. but the node shutdown management is controlled by different feature-gates, GracefulNodeShutdown for linux and WindowsGracefulNodeShutdown for windows. Due to the implementation/design order, the GracefulNodeShutdown does not have the `Linux` prefix.

Reply:
Is there any code that prevents running kubelet on Linux with enabled `WindowsGracefulNodeShutdown` feature gate?

Reply:
> Is there any code that prevents running kubelet on Linux with enabled `WindowsGracefulNodeShutdown` feature gate?

There shouldn't be, because we intend to graduate this feature to GA!

Reply:
So, kubelet can be run with WindowsGracefulNodeShutdown on Linux and below code will be executed when it shuldn't, right?

Reply:
@bart0sh  I change the code according to @sftim [suggestions](https://github.com/kubernetes/kubernetes/pull/129245#discussion_r1892364820).